ext/har/logger.go

```diff
@@ -0,0 +1,157 @@
+package har
+
+import (
+    "net/http"
+    "time"
+    "github.com/elazarl/goproxy"
+)
+
+// ExportFunc is a function type that users can implement to handle exported entries
+type ExportFunc func([]Entry)
+
+// Logger implements a HAR logging extension for goproxy
+type Logger struct {
+    entries        []Entry
+    captureContent bool
+    exportFunc      ExportFunc
+    exportInterval  time.Duration
+    exportThreshold int
+    stopCh          chan struct{}
+    dataCh          chan Entry
+}
+
+// LoggerOption is a function type for configuring the Logger
+type LoggerOption func(*Logger)
+
+// WithExportInterval sets the interval for automatic exports
+func WithExportInterval(d time.Duration) LoggerOption {
+    return func(l *Logger) {
+        l.exportInterval = d
+    }
+}
+
+// WithExportCount sets the number of requests after which to export entries
+func WithExportThreshold(threshold int) LoggerOption {
+    return func(l *Logger) {
+        l.exportThreshold = threshold
+    }
+}
+
+// NewLogger creates a new HAR logger instance
+func NewLogger(exportFunc ExportFunc, opts ...LoggerOption) *Logger {
+    l := &Logger{
+        entries:        make([]Entry, 0),
+        captureContent: true,
+        exportFunc:     exportFunc,
+        exportThreshold: 100,    // Default threshold
+        exportInterval: 0,       // Default no interval
+        stopCh:         make(chan struct{}),
+        dataCh:         make(chan Entry, 1000), 
```

Comment:
```suggestion
        dataCh:         make(chan Entry), 
```

Use a normal channel, not a buffered channel, here it's enough